### ISSUE TITLE
Files Are Randomly Overwritten With Contents of Other Files When Saving

### ISSUE THREAD

**Issue by pahund** [CompuIves/codesandbox-client#1020]
# 🐛 bug report

## Description of the problem

While working with CodeSandbox, when switching between files in the file editor, sometimes it happens that the content of the file being switched to are replaced with the contents of the file that was previously open.

It seems to happen randomly, but frequently. I've observed the bug with Chrome 67 and Firefox 62, both on MacOS 10.13.6.

It may have something to do with saving the files using the “Save” button in the top menu, possibly in conjunction with Prettier, which I use frequently, invoked by keyboard shortcut.

I've uploaded a video that shows me reproducing the issue: 

https://youtu.be/51MpKqLBOJU

At 2:15, I hit the save button, at 2:18, the contents of App.js and Hello.js are suddenly identical.

## How has this issue affected you? What are you trying to accomplish?

Since the code change cannot be undone, it makes it very hard and unnerving for me to work productively with CodeSandbox. I've resorted to downloading the code frequently to have a backup of my code when the bug happens.

###  Link to sandbox: 

Here's a sandbox where the bug caused the code from one module to magically replace the code of two others, so that App.js, Hello.js and StartEditing.js all have the same contents: 
https://codesandbox.io/s/7m52okr8yx

### Your Environment
| Software         | Name/Version|
| ---------------- | ---------- |
| Сodesandbox  | PROD-1532796593-1ab3af9 |
| Browser | Chrome 67, Firefox 62 |
| Operating System | MacOS 10.13.6 |

**Comment by deadcoder0904** [comment]
Same issue for me. The most important feature is **Save Files** & its broken now. I used CodeSandBox  sometime 12 hours ago but the new changes broke the Save Functionality. Can't do anything.

**Comment by CompuIves** [comment]
This is the best issue description I've had in a while! Thank you, I will work on a fix today.

**Comment by CompuIves** [comment]
Just pushed a fix for it. Will deploy in within the hour!

**Comment by pahund** [comment]
👍🏻Thank you, keep up the great work!

**Comment by deadcoder0904** [comment]
Well its not working yet

**Comment by pahund** [comment]
It works for me, I cannot reproduce the bug anymore (MacOS 10.13.6, Firefox Dev Edition 62

**Comment by deadcoder0904** [comment]
Okay so now it worked after I deleted the Local Storage & Session Storage.

**Comment by deadcoder0904** [comment]
Thank you @CompuIves 🙌

**Comment by siordache94** [comment]
Issue isn't solved 
what information can I provide?

**Comment by neonovic** [comment]
Is this resolved? I am new to Codesandbox, but I had one file overwritten by another and I don't know why.

**Comment by bj-vishal** [comment]
> Okay so now it worked after I deleted the Local Storage & Session Storage.

how to delete the local storage and session storage.

**Comment by CharlesDerek** [comment]
I encountered the same issue, which was resolved by clearing session and local storage. The problem arises when opening two sandboxes in different tabs or windows of the same browser.

The application functions as a Single Page Application (SPA) and experiences conflicts when both tabs write to local storage, leading to a race condition and inconsistencies.

For users needing to compare two sandbox IDEs side by side, opening one of the instances in an incognito window can prevent these conflicts by isolating storage.

For the development team, a potential solution would be to implement unique storage keys for each instance. Alternatively, using session storage for instance-specific data could also mitigate these conflicts.

![Image](https://github.com/user-attachments/assets/aa5568e4-f1ca-47e8-91ed-0c802b88b2f9)
